src/commands/EmbedCommand.ts

```diff
@@ -0,0 +1,87 @@
+import { Command } from "../interactions/interactionClasses";
+import {
+	ActionRowBuilder,
+	ChatInputCommandInteraction,
+	ModalBuilder,
+	SlashCommandBuilder,
+	TextInputBuilder,
+	TextInputStyle,
+} from "discord.js";
+
+class EmbedCommand extends Command {
+	constructor() {
+		super("embed");
+	}
+
+	async execute(interaction: ChatInputCommandInteraction): Promise<void> {
+		const fields =
+			interaction.options
+				.getString("fields")
+				?.split(",")
+				.map((s) => s.trim())
+				.filter((s) => s)
+				.slice(0, 3) ?? [];
+		const showAuthor = interaction.options.getBoolean("show-author") ?? false;
+
```

Comment:
Could check that fields titles are at most 256 characters. Though custom ids have a max length of 100 characters. So they can't fit long field titles. Modals are pretty temporary so maybe could just add a random key in the custom id and just reference a in-memory map.